syntax-analysis.py

```diff
@@ -12,39 +20,70 @@ def fileAnalysis(file):
     printKeywords = 0 # holds the number of times the print function is called
     while line is not None:
         if sub == 0:
-            if line.__contains__('def'):
+            if line.__contains__('def '): # WE WILL NEED TO ADD FUNCATIONALITY TO CHECK IF THERE IS NO SPACE THIS IS A TEMP FIX
                     if line.__contains__('(') and line.__contains__(')') and line.__contains__(':'):
                         sub +=1
+                        out.write(line)
                         line = opened.readline()
                     elif line.find('(') == -1: # this denotes the the first paraenthesis for the def function is missing
                         # since strings are immutatable have to make a new string with the missing piece and then overwrite that line of the file
                         line = line[:line.find('def') + 3] + '(' + line[line.find('def') + 3:]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     elif line.find(')') == -1:
                         line = line[:line.find(':')] + ')' + line[line.find(':'):]
                         out.write(line)
-                        opened.readline()
+                        line = opened.readline()
                     else:
                         line = line + ':'
                         out.write(line)
-                        opened.readline()
-            elif line.__contains__('if'):
+                        line = opened.readline()
+            elif line.__contains__('if '):
                 if line.__contains__(':'):
                     sub +=1
+                    out.write(line)
                     line = opened.readline()
                 else:
                     line = line + ':'
                     out.write(line)
-                    opened.readline()
+                    line = opened.readline()
             else:
                 printKeywords += line.count('print(') # Checking for the first paraenthesis since only function calls are wanted, not the actual word
                 line = opened.readline()
+        else:
+            if line.__contains__('def '):
+                if line.__contains__('(') and line.__contains__(')') and line.__contains__(':'):
+                    line, sub = indentationCheck(line, sub) # Example using new indentation check method and skeleton for rest
+                    out.write(line)
+                    line = opened.readline()
+                elif line.find('(') == -1: 
+                    line = line[:line.find('def') + 3] + '(' + line[line.find('def') + 3:]
+                    # Input measures for the indentation check
+                elif line.find(')') == -1:
+                    line = line[:line.find(':')] + ')' + line[line.find(':'):]
+                    # Input measures for indentation check
+                else:
```

Comment:
the same comment mentioned earlier, this case will never be met.
Check to make sure only one element is missing at a time so the else case is met correctly.